docs/fsharp/whats-new/fsharp-9.md

```diff
@@ -0,0 +1,514 @@
+
+Previously, you had:
+
+```fsharp
+#nowarn "0070"
+#time "on"
+```
+
+Now, you can write:
+
+```fsharp
+#nowarn 0070
+#time on
+```
+
+This also ties into the next two changes.
+
+## Extended #help directive in fsi to show documentation in the REPL
+
+The `#help` directive in F# Interactive now shows documentation for a given object or function, which you can now pass without quotes.
+
+```
+> #help List.map;;
+
+Description:
+Builds a new collection whose elements are the results of applying the given function
+to each of the elements of the collection.
+
+Parameters:
+- mapping: The function to transform elements from the input list.
+- list: The input list.
+Returns:
+The list of transformed elements.
+
+Examples:
+let inputs = [ "a"; "bbb"; "cc" ]
+
+inputs |> List.map (fun x -> x.Length)
+// Evaluates to [ 1; 3; 2 ]
+
+Full name: Microsoft.FSharp.Collections.ListModule.map
+Assembly: FSharp.Core.dll
+```
+
+See [Enhancing #help in F# Interactive blog post](https://devblogs.microsoft.com/dotnet/enhancing-help-in-fsi/) for more details.
+
+## Allow #nowarn to support the FS prefix on error codes to disable warnings
+
+Previously, when you wanted to disable a warning and wrote `#nowarn "FS0057"`, you would get an `Invalid warning number 'FS0057'`. Even though the warning number is correct, it just wasn't supposed to have the `FS` prefix.
+
+Now, you won't have to spend time figuring that out because the warning numbers are accepted even with the prefix.
+
+All of these will now work:
+
+```fsharp
+#nowarn 57
+#nowarn 0057
+#nowarn FS0057
+
+#nowarn "57"
+#nowarn "0057"
+#nowarn "FS0057"
+```
+
+It's a good idea to use the same style throughout your project.
+
+## Warning about TailCall attribute on non-rec functions or let-bound values
+
+F# now emits a warning when you put the `[<TailCall>]` attribute somewhere it doesn't belong. While it has no effect on what the code does, it could confuse someone reading it.
+
+For example, these usages will now emit a warning:
+
+```fsharp
+[<TailCall>]
+let someNonRecFun x = x + x
+
+[<TailCall>]
+let someX = 23
+
+[<TailCall>]
+let rec someRecLetBoundValue = nameof(someRecLetBoundValue)
+```
+
+## Enforce attribute targets
+
+The compiler now correctly enforces the `AttributeTargets` on let values, functions, union case declarations, implicit constructors, structs, and classes. This can prevent some hard-to-notice bugs, such as forgetting to add the unit argument to an Xunit test.
+
+Previously, you could write:
+
+```fsharp
+[<Fact>]
+let ``this test always fails`` =
+  Assert.True(false)
+```
+
+When you ran the tests with `dotnet test`, they would pass. Since the test function is not actually a function, it was ignored by the test runner.
+
+Now, with correct attribute enforcement, you will get an `error FS0842: This attribute is not valid for use on this language element`.
+
+## Updates to the standard library (FSharp.Core)
```

Comment:
Should we add a link to the FSharp.Core library [Nuget](https://www.nuget.org/packages/FSharp.Core/) or docs]?